### DISCUSSION TITLE
how are sessions tracked? do they expire?

### DISCUSSION THREAD

**Discussion by joeyscarim** [umami-software/umami#1681] [Q&A]
i’m trying to understand how the sessions response on the /pageviews endpoint works, but i’m not understanding what defines a session. 

if a visitor visits the site on the same device (same user agent and same ip) 10 days in a row, and then i get the session count for those 10 days, would that only count as 1 session?

thanks for any insight

**Comment by LunyaaDev** [comment]
From my understanding the sessionID is a [hash](https://github.com/umami-software/umami/blob/88da20ea7f9e34a3d09cf8503ae09bff63b254bc/src/lib/session.ts#L70), that is defined by [`website` and `hostname`](https://github.com/umami-software/umami/blob/88da20ea7f9e34a3d09cf8503ae09bff63b254bc/src/lib/session.ts#L46) send in the POST body, the [`ip` and `userAgent`](https://github.com/umami-software/umami/blob/88da20ea7f9e34a3d09cf8503ae09bff63b254bc/src/lib/session.ts#L67) and the [UTCString of the beginning of the month](https://github.com/umami-software/umami/blob/88da20ea7f9e34a3d09cf8503ae09bff63b254bc/src/lib/crypto.ts#L9), which gets added wenn calling the [hashing](https://github.com/umami-software/umami/blob/88da20ea7f9e34a3d09cf8503ae09bff63b254bc/src/lib/crypto.ts#L21) function.


The [`useSession`](https://github.com/umami-software/umami/blob/88da20ea7f9e34a3d09cf8503ae09bff63b254bc/src/lib/middleware.ts#L30) middleware adds the session to the request.

Which  then gets used by the [`/api/send`](https://github.com/umami-software/umami/blob/88da20ea7f9e34a3d09cf8503ae09bff63b254bc/src/pages/api/send.ts#L106) endpoint.


> if a visitor visits the site on the same device (same user agent and same ip) 10 days in a row, and then i get the session count for those 10 days, would that only count as 1 session?

If all the visits are in the same month, then yes.
